src/wifi/model/mac-low.cc

```diff
@@ -3055,14 +3042,31 @@ MacLow::StopMpduAggregation (Ptr<const Packet> peekedPacket, WifiMacHeader peeke
       aPPDUMaxTime = MicroSeconds (10000);
     }
 
+  WifiMacHeader hdr = m_currentHdr;
+  MacLowTransmissionParameters txParams = m_txParams;
+  uint32_t peekedSize = peekedPacket->GetSize () + peekedHdr.GetSize () + WIFI_MAC_FCS_LENGTH;
+  if (aggregatedPacket->GetSize () > edcaIt->second->GetMsduAggregator ()->GetMaxAmsduSize ())
+    txParams.EnableCompressedBlockAck ();
+  else
+    txParams.EnableAck ();
```

Comment:
missing {}

Reply:
also, i'm not sure about this. you're modifying copy of m_txParams inside StopMpduAggregation, how can you be sure that this will also happen from the calling function?

Reply:
my previous comment is valid also here